### PR TITLE
[#3963][followup] feat(core): Call the method ownerset of authorization plugin

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?

This is the follow-up pull request of #3963.
 #3963 add authorization plugin to set owner. This pull request is to call them.

### Why are the changes needed?

Fix: #3963

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Add new ut.

### CODE REVIEW

File: core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java [hunk 1]
```diff
@@ -190,6 +207,10 @@ public static boolean needApplyAllAuthorizationPlugin(SecurableObject securableO
     return false;
   }
 
+  private static boolean needApplyAuthorizationPluginSetOwner(MetadataObject.Type type) {
+    return type == MetadataObject.Type.METALAKE;
+  }
+
   private static boolean supportsSingleAuthorizationPlugin(MetadataObject.Type type) {
```
Comment (xunliu) [inline core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java:214]: What does `Single` mean in the `supportsSingleAuthorizationPlugin()` function?
Can we have a better name?

File: core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java [hunk 2]
```diff
@@ -150,29 +150,46 @@ public static void callAuthorizationPlugin(
...
+          callAuthorizationPluginImpl(consumer, catalog);
+        }
+      }
+    }
+  }
+
+  public static void callAuthorizationPluginForMetadataObject(
+      String metalake, MetadataObject metadataObject, Consumer<AuthorizationPlugin> consumer) {
+    CatalogManager catalogManager = GravitinoEnv.getInstance().catalogManager();
+    if (needApplyAuthorizationPluginSetOwner(metadataObject.type())) {
```
Comment (xunliu) [inline core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java:172]: I think we can directly judge `if (metadataObject.type() == MetadataObject.Type.METALAKE)` would be clear.

File: core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java (hunk 1 above)
Comment (jerqi) [reply core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java:214]: I changed to `supportsPushdownAuthorizationPlugin`.

File: core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java (hunk 2 above)
Comment (jerqi) [reply core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java:172]: I changed to `needApplyAllAuthorizationPlugin`

File: core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java [hunk 3]
```diff
@@ -190,7 +207,11 @@ public static boolean needApplyAllAuthorizationPlugin(SecurableObject securableO
     return false;
   }
 
-  private static boolean supportsSingleAuthorizationPlugin(MetadataObject.Type type) {
+  private static boolean needApplyAllAuthorizationPlugins(MetadataObject.Type type) {
```
Comment (xunliu) [inline core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java:210]: Change `needApplyAllAuthorizationPlugins()` to `needApplyAuthorizationAllCatalogs()` ?

Comment (jerqi) [reply core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java:210]: OK. Done.

**Review by xunliu — APPROVED** [review]
LGTM

**Pull request merged**